pulsar-broker/src/main/java/org/apache/pulsar/broker/service/ServerCnx.java

```diff
@@ -817,6 +817,17 @@ protected void handleConnect(CommandConnect connect) {
             return;
         }
 
+        if (!this.service.getPulsar().isRunning()) {
```

Comment:
If `!service.isAuthenticationEnabled()` return true, the connect will complete without checking pulsar service  is running or not.

 My suggestion is that move this code block just to Line807,  which is just below the debug log @HQebupt

Reply:
Resolved, thank you. PTAL @AnonHxy @congbobo184

Reply:
LGTM now